test/packages/parallel/ti_anomali_logsdb/docs/README.md

```diff
@@ -137,7 +137,6 @@ An example event for `threatstream` looks as following:
         }
     }
 }
-
```

Comment:
ISTM that a separate PR in integrations that does
```
parallel "perl -p -i -e 'chomp if eof'" {} ::: $(rg -l -U -g 'sample_event.json' '^}\n') 
for d in $(git diff --name-only|cut -f1,2 -d/|sort|uniq); do (cd $d; elastic-package build); done
```
would fix all the cases to pass in both versions of elastic-package. This would be done prior to bumping the ep version for integrations.

Reply:
> would fix all the cases to pass in both versions of elastic-package. This would be done prior to bumping the ep version for integrations.

Would this be compatible with the current version of elastic-package?

Reply:
Yes, that mutation takes the state of the packages to what they would be without human intervention.

Reply:
I mean, this change breaks linting on current integrations, and applying the fix in current integrations with current elastic-package will also break linting.

We should find a way of keeping compatibility with current integrations. When we needed to do this kind of changes we have done them in a way that depend on a spec version, so updating the spec version is what triggers the new behaviour. Or we have kept compatibility with both formats.

Reply:
> I mean, this change breaks linting on current integrations, and applying the fix in current integrations with current elastic-package will also break linting.

Why do you say that?

If the fix I show above is run, it makes no difference to the output of the linter. After it is run, the change in this PR makes no difference to the output of the linter.

Reply:
>> I mean, this change breaks linting on current integrations, and applying the fix in current integrations with current elastic-package will also break linting.
>
> Why do you say that?

If we apply this change in elastic-package, we break linting in current packages, as found in https://github.com/elastic/integrations/pull/11055.

If we apply the fix to current integrations, before making the change in elastic-package, then linting also breaks, tried in https://github.com/elastic/integrations/pull/11170.

So it looks like there is no way to apply this change in elastic-package and the one in integrations independently. We need to apply this change in elastic-package and remove the empty lines in the integrations repository _at the same time_. And once we do that, linting will fail for old versions of packages.

In general we try to avoid this kind of changes, they can be problematic with current PRs, local environments, backports and so on.

> IMO the mutation above should be run anyway since it corrects incorrect human edits to files that should not have been made.

I am not opposed to fix this, it makes sense to me, but if possible we should do it in a non-breaking way.

Maybe I am misunderstanding something. Is there any way to apply both fixes independently? A PR with the proposed fix in integrations might help.

Reply:
https://github.com/elastic/integrations/pull/11170 is not what the code above does.

Reply:
Do we need to have changelog entries for packages if we are just fixing these test defects?

Reply:
This is the change https://github.com/elastic/integrations/pull/11174.

Reply:
> Do we need to have changelog entries for packages if we are just fixing these test defects?

No need for this (unless some team wants this change to be published).

> This is the change [elastic/integrations#11174](https://github.com/elastic/integrations/pull/11174).

Thanks, this helps.

Looking at the needed changes, what would you think about applying the fix in elastic-package only to the methods that write the events, and not to `renderSampleEvent`? This way we would keep compatibility with new and old packages.

Reply:
> Looking at the needed changes, what would you think about applying the fix in elastic-package only to the methods that write the events, and not to renderSampleEvent? This way we would keep compatibility with new and old packages.

No, this doesn't make sense as we are adding the newline in one place and removing in the other.

Reply:
What I think confuses me from https://github.com/elastic/integrations/pull/11174 is that we are removing the newlines, but here we are adding them, looks like doing the opposite of what we are intending here.

Reply:
This is code repair. The ultimate goal is to have a state of code that is agnostic to the presence of final new lines in the sample_event.json files. This does that. The change in elastic/integrations#11174 works around (by fixing the packages to conform exactly to what ep emits) the weaknesses in the linting code in ep. The new code here does not care about whether the sample_event.json file has a final new line, it will always render the event without any trailing whitespace and add the necessary newline when required.

The deeper background for this (and some of the correctness issues) can be see [here](https://github.com/elastic/integrations/pull/10811#discussion_r1746144799).